Lib/asyncio/selector_events.py

```diff
@@ -921,14 +925,74 @@ def write(self, data):
             self._loop._add_writer(self._sock_fd, self._write_ready)
 
         # Add it to the buffer.
-        self._buffer.extend(data)
+        self._buffer.append(data)
```

Comment:
Using `_buffer.append()` instead of `_buffer.extend()` here seems like it would prevent passing an iterable as an argument for *data* for `_SelectorSocketTransport.write()`.